src/main/java/com/google/sps/servlets/data/util/GmapsHelper.java

```diff
@@ -0,0 +1,150 @@
+   */
+  public CategoryGroup getCategoryGroupById(int Id) {
+    switch (Id) {
+      case 2:
+        return CategoryGroup.RECREATION;
+      case 3:
+        return CategoryGroup.DINING;
+      case 4:
+        return CategoryGroup.GROCERY;
+      default:
+        return CategoryGroup.GROCERY;
+    }
+  }
+
+  /**
+   * Searches for places within `radiusMeters` nearby `anchorLatitude` and `anchorLongitude`
+   *
+   * @param categoryGroup: categoryGroup for which to search
+   * @param anchorLatitude: The latitude coordinate of the location to search around
+   * @param anchorLongitude: The longitude coordinate of the location to search around
+   * @param radiusMeters: The radius, in meters, to search around the specified `anchorLatitude` and
+   *     `anchorLongitude`
+   * @return A valid JSON string containing the results
+   */
+  public String searchNearbyCategoryGroup(
+      CategoryGroup categoryGroup, double anchorLatitude, double anchorLongitude, int radiusMeters)
+      throws ApiException, InterruptedException, IOException {
+    PlaceType[] placeTypes;
+    switch (categoryGroup) {
+      case RECREATION:
+        placeTypes = this.recreationTypes;
+        break;
+      case DINING:
+        placeTypes = this.diningTypes;
+        break;
+      case GROCERY:
+        placeTypes = this.groceryTypes;
+        ;
+        break;
+      default:
+        placeTypes = new PlaceType[0];
+    }
+
+    PlacesSearchResult[] results =
+        this.nearbySearchMultipleType(placeTypes, anchorLatitude, anchorLongitude, radiusMeters);
+    Gson gson = new Gson();
+    return gson.toJson(results);
+  }
+
+  private PlacesSearchResult[] nearbySearchMultipleType(
+      PlaceType[] types, double anchorLatitude, double anchorLongitude, int radiusMeters)
+      throws ApiException, InterruptedException, IOException {
+    PlacesSearchResult[] accumulatedResults = new PlacesSearchResult[0];
+    for (PlaceType type : types) {
+      PlacesSearchResult[] iterationResults =
+          this.nearbySearchByPlaceType(type, anchorLatitude, anchorLongitude, radiusMeters);
+      accumulatedResults =
+          Arrays.copyOf(accumulatedResults, accumulatedResults.length + iterationResults.length);
+      System.arraycopy(
+          iterationResults,
+          0,
+          accumulatedResults,
+          (accumulatedResults.length - iterationResults.length),
+          iterationResults.length);
+    }
+    return accumulatedResults;
+  }
+
+  public PlacesSearchResult[] nearbySearchByPlaceType(
+      PlaceType placeType, double anchorLatitude, double anchorLongitude, int radiusMeters)
+      throws ApiException, InterruptedException, IOException {
+    LatLng location = new LatLng(anchorLatitude, anchorLongitude);
+    PlacesSearchResponse response =
+        PlacesApi.nearbySearchQuery(this.context, location)
+            .radius(radiusMeters)
+            .rankby(RankBy.PROMINENCE)
+            .type(placeType)
+            .language("en")
```

Comment:
It's fine we only support English for now. Let's a create issue to track localization